File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.java
class Solution {
    public int[] solution(int n, int[] slicer, int[] num_list) {
        int[] answer = {};
        
        int a = slicer[0];
        int b = slicer[1];
        int c = slicer[2];
        
        int idx = 0;
        
        if(n == 1){
            answer = new int[b+1];
            for(int i=0; i<=b; i++){
                answer[idx++] = num_list[i];
            }
        }else if(n == 2){
            answer = new int[num_list.length - a];
            for(int i=a; i<num_list.length; i++){
                answer[idx++] = num_list[i];
            }
        }else if(n == 3){
            answer = new int[b - a + 1];
            for(int i=slicer[0]; i<=b; i++){
                answer[idx++] = num_list[i];
            }
        }else{
            if((b - a) % 2 == 0){
                answer = new int[(b - a + 1) / c + 1];
            }else{
                answer = new int[(b - a + 1) / c];
            }
            for(int i=slicer[0]; i<=b; i+=c){
                answer[idx++] = num_list[i];
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181897. 리스트 자르기/리스트 자르기.kt
class Solution {
    fun solution(n: Int, slicer: IntArray, num_list: IntArray): IntArray {
        var answer: IntArray = intArrayOf()
        
        var list: List<Int> = num_list.asList()
        
        if(n == 1){
            list = list.slice(0 .. slicer[1])
        }else if(n == 2){
            list = list.slice(slicer[0] .. list.size-1)
        }else if(n == 3){
            list = list.slice(slicer[0] .. slicer[1])
        }else{
            list = list.slice(slicer[0] .. slicer[1] step slicer[2])
        }
        
        answer = list.toIntArray()
        return answer
    }
}